### PR TITLE
Updates for Julia 0.6

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

File: src/GLM.jl [hunk 1]
```diff
@@ -54,14 +55,14 @@ module GLM
         updateμ!,      # update the response type from the linear predictor
         wrkresp         # working response
 
-    typealias FP AbstractFloat
-    typealias FPVector{T<:FP} AbstractArray{T,1}
+    const FP = AbstractFloat
+    @compat const FPVector{T<:FP} = AbstractArray{T,1}
```
Comment (nalimilan) [inline src/GLM.jl]: Technically I think `const` isn't needed, but that's so subtle that I don't think we care.

File: src/linpred.jl [hunk 2]
```diff
@@ -15,20 +15,19 @@ function installbeta!(p::LinPred, f::Real=1.)
...
-typealias BlasReal Union{Float32,Float64}
-
 type DensePredQR{T<:BlasReal} <: DensePred
     X::Matrix{T}                  # model matrix
     beta0::Vector{T}              # base coefficient vector
     delbeta::Vector{T}            # coefficient increment
     scratchbeta::Vector{T}
     qr::QRCompactWY{T}
-    function DensePredQR(X::Matrix{T}, beta0::Vector{T})
+    @compat function (::Type{DensePredQR{T}}){T}(X::Matrix{T}, beta0::Vector{T})
```
Comment (nalimilan) [inline src/linpred.jl]: Is `@compat` really needed? Same below.

Comment (andreasnoack) [inline src/linpred.jl]: I don't think it is

Comment (nalimilan) [inline src/linpred.jl]: Then please remove it before merging, these annotations are annoying. :-)

Comment (ararslan) [inline src/linpred.jl]: I thought it was needed for the type parameter on `new`?

File: src/GLM.jl (hunk 1 above)
Comment (ararslan) [inline src/GLM.jl]: I think it's the preferred style, as it's what's suggested by the deprecation warning and is what's used in Base.

File: src/linpred.jl (hunk 2 above)
Comment (ararslan) [inline src/linpred.jl]: Turns out I was wrong--not needed for `new{T}` on 0.5. ¯\\\_(ツ)\_/¯

**Pull request merged**